gensim/similarities/nmslib.py

```diff
@@ -210,21 +219,23 @@ def _build_from_model(self, vectors, labels):
         self.labels = labels
 
     def most_similar(self, vector, num_neighbors):
-        """Find the approximate `num_neighbors` most similar items.
+        """Find `num_neighbors` most similar items, using NMSLIB's approximate algorithms.
 
         Parameters
         ----------
         vector : numpy.array
-            Vector for word/document.
+            Vector for a word or document.
         num_neighbors : int
-            Number of most similar items
+            How many most similar items to look for?
 
         Returns
         -------
         list of (str, float)
-            List of most similar items in format [(`item`, `cosine_distance`), ... ]
+            List of most similar items in the format `[(item, cosine_similarity), ... ].
 
         """
         ids, distances = self.index.knnQueryBatch(vector.reshape(1, -1), k=num_neighbors)[0]
 
-        return [(self.labels[ids[i]], 1 - distances[i] / 2) for i in range(len(ids))]
+        # NMSLIB returns cosine distance (not similarity), which is simply `dist = 1 - cossim`.
+        # So, convert back to similarities here.
+        return [(self.labels[ids[i]], 1 - distances[i]) for i in range(len(ids))]
```

Comment:
Do we need tests to cover this fix?

Reply:
Maybe. I'm waiting for the CI to complete, to see what tests failed :)

Let me try to find where / if nmslib is being tested.

Reply:
Do want to deal with this on the develop branch instead?

If this is not necessary for 3.8.3, then I'd rather release 3.8.3 without it, because it has already taken long enough. The alternative is:

1. Add tests
2. Merge onto develop
3. Merge develop into release-3.8.3 (we have done it at least once in the past)
4. Wait for CI to complete
5. Build wheels
6. Release

That will add at least another day to the release.

Reply:
No tests failed – we're testing the returned ids, but not similarities.

I think this is on purpose, because the model is trained on a tiny corpus and the exact vectors are probably unstable => similarities will differ between runs.

Do you think it's worthwhile changing the tests to be deterministic, and then testing for exact similarities too?

Reply:
> Do want to deal with this on the develop branch instead?

Like I said up to you – nothing in this PR is critical. Just another little clean up.

Reply:
> /Volumes/work/workspace/vew/tst/bin/ipython:8: DeprecationWarning: Call to deprecated `most_similar` (Method will be removed in 4.0.0, use self.wv.most_similar() instead).
  sys.exit(start_ipython())
Out[2]: [('cat', 1.000000238418579), ('woof', 0.12982195615768433)]